### PR TITLE
Add mobile data sensors

### PR DESCRIPTION
**Pull request by frimtec**
## Summary
New mobile data sensors to check whether some mobile data settings are enabled or not.

## Screenshots
Screenshot_Dark
Screenshot_Light

## Link to pull request in Documentation repository
Documentation: home-assistant/companion.home-assistant#435

## Any other notes
Used to remind family members to switch of mobile data when they are back home to save costs for prepaid data subscription.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @frimtec, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;android#1317).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -48,6 +48,8 @@
   <string name="basic_sensor_name_location_accurate">Einzelne genaue Position</string>
   <string name="basic_sensor_name_location_background">Hintergrund Standort</string>
   <string name="basic_sensor_name_location_zone">Standortzone</string>
+  <string name="basic_sensor_name_mobile_data">Mobile Daten</string>
```
Comment (dshokouhi) [inline app/src/main/res/values-de/strings.xml:51]: Please remove these strings from this time. We use lokalize which updates these strings. Only English needs strings here.

**Review by dshokouhi — CHANGES_REQUESTED** [review]
Please make use of `hasSensor` function to only show these sensors on devices that actually have a cellular modem. Devices that are WiFi only should not see these sensors. Please also only update English and remove the additional language strings as this hey updated from lokalize.

File: app/src/main/java/io/homeassistant/companion/android/sensors/MobileDataManager.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        context: Context,
+        sensor: SensorManager.BasicSensor,
+        settingKey: String,
+        icon: String
+    ) {
+        if (!isEnabled(context, sensor.id))
+            return
+
+        var enabled = false
+        if (checkPermission(context, sensor.id)) {
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/sensors/MobileDataManager.kt:55]: What permission are we checking here? `requiredPermissions` is blank up above.

Comment (frimtec) [reply app/src/main/java/io/homeassistant/companion/android/sensors/MobileDataManager.kt]: Removed obsolete call to checkPermission.

File: app/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (frimtec) [reply app/src/main/res/values-de/strings.xml]: Done.

**Comment by frimtec** [comment]
Following findings resolved:
* Removed German translations.
* Implemented ```hasSensor```.
* Remove obsolete call ```checkPermission```.

File: app/src/main/java/io/homeassistant/companion/android/sensors/SensorReceiver.kt [hunk 3]
```diff
@@ -47,7 +47,8 @@ class SensorReceiver : BroadcastReceiver() {
             StepsSensorManager(),
             StorageSensorManager(),
             TimeZoneManager(),
-            TrafficStatsManager()
+            TrafficStatsManager(),
+            MobileDataManager()
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/sensors/SensorReceiver.kt:51]: Should we consider keeping this list alphabetized?

Comment (frimtec) [reply app/src/main/java/io/homeassistant/companion/android/sensors/SensorReceiver.kt]: Sure, sorry for that, it is fixed now.

**Review by dshokouhi — APPROVED** [review]
Looks good to me, thanks for your contribution!

**Comment by frimtec** [comment]
Many thanks for your review.

**Review by JBassett — APPROVED** [review]

**Pull request merged**